tests/lib/edge_impulse/src/edge_impulse_zip/edge-impulse-sdk/cmake/utils.cmake

```diff
@@ -0,0 +1,11 @@
+# Macro that is used by the CMake while building Edge Impulse library
+
+MACRO(RECURSIVE_FIND_FILE return_list dir pattern)
+    FILE(GLOB_RECURSE new_list "${dir}/${pattern}")
```

Comment:
Why using a glob pattern here ?
Do you have that many files.

https://cmake.org/cmake/help/latest/command/file.html#glob-recurse
> **Note:** We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.

Reply:
It was done mainly to keep the required components of the mocked zip file consistent with zip file deployed and downloaded from Edge Impulse studio. I will rework it, as mentioned in the comment above.

Reply:
Modifying the way how we build the library must be aligned with Edge Impulse team to ensure that zip file content will remain proper.

Reply:
Ahh, I see the build will use the template from here:
https://github.com/nrfconnect/sdk-nrf/blob/master/lib/edge_impulse/CMakeLists.ei.template

```
[6/10] cd /projects/github/ncs/nrf/tests/lib/edge_impulse/build/edge_impulse/src/edge_impulse_project && /opt/cmake-3.21.3-linux-x86_64/bin/cmake -E copy /projects/github/ncs/nrf/lib/edge_impulse/CMakeLists.ei.template /projects/github/ncs/nrf/tests/lib/edge_impulse/build/edge_impulse/src/edge_impulse_project/CMakeLists.txt
```

Could you please add a comment in this file so that it is clear for the reader why this file is needed and how it's being used.